app/controllers/api/providers_controller.rb

```diff
@@ -91,7 +91,12 @@ def format_provider_custom_attributes(attribute)
         attribute["value"] = attribute.delete("field_type").safe_constantize.parse(attribute["value"])
       end
       attribute["section"] ||= "metadata" unless @req.action == "edit"
+      if attribute["section"].present? && !(CustomAttribute::ALLOWED_API_SECTIONS.include? attribute["section"])
+        raise ForbiddenError, "Unauthorized attribute section specified: #{attribute["section"]}"
```

Comment:
this should be a BadRequestError, ForbiddenError is meant for HTTP 403 (i.e. resource RBAC) maybe change to:

```
raise "Invalid attribute section specified: #{attribute["section"]}"
```

and remove the rescue ForbiddenErrror below, it will be picked up by the rescue and raise in lines 100-101.
```